compiler/builtins/string.ts

```diff
@@ -1275,7 +1275,43 @@ i32.store8 0 12`;
       Porffor.wasm.i32.store16(Porffor.wasm`local.get ${tmp}` + tmpLen * 2, x, 0, 4);
       tmpLen++;
     }
-  } else {
+  } 
+  else if (sepLen === 0) {
+    const clammedLimit: i32 = limit > thisLen ? thisLen : limit;
+    for (let i = 0; i <= clammedLimit; i+=2) {
+      tmp = Porffor.allocate();
```

Comment:
A full allocate is very costly, could you just do `Porffor.allocateBytes(8)` (sorry if some other code in this file does this, that was before this existed and needs updating)

Reply:
Yeah, i had very little idea of what code was doing; I saw you were copying chars to `temp` in each loop, thats why